pkg/oci/oci.go

```diff
@@ -871,6 +876,17 @@ func ensureImage(ctx context.Context, imageStore oras.Target, image string, imgO
 		}
 	}
 
+	if imgOpts.Registry != "" {
+		targetImage, err := normalizeImageName(image)
+		if err != nil {
+			return fmt.Errorf("normalizing image: %w", err)
+		}
+
+		if !strings.HasPrefix(targetImage.String(), imgOpts.Registry) {
+			return fmt.Errorf("%s was not pulled from %s", targetImage, imgOpts.Registry)
```

Comment:
```suggestion
			return fmt.Errorf("%s not originating from allowed registry %s", targetImage, imgOpts.Registry)
```

"not pulled from" sounds like the actual pulling failed.